packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -0,0 +1,216 @@
+
+const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+	[rootDataObjectFactory.type, Promise.resolve(rootDataObjectFactory)],
+	[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+]);
+const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+	rootDataObjectFactory,
+	registryStoreEntries,
+	undefined,
+	[],
+	runtimeOptions,
+);
+
+async function createSummarizer(
+	provider: ITestObjectProvider,
+	container: IContainer,
+	summaryVersion?: string,
+) {
+	const summarizerRuntimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+		rootDataObjectFactory,
+		registryStoreEntries,
+		undefined,
+		[],
+		{ summaryOptions: defaultSummaryOptions },
+	);
+	const loader = provider.createLoader(
+		[[provider.defaultCodeDetails, summarizerRuntimeFactory]],
+		{
+			configProvider: mockConfigProvider(),
+		},
+	);
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+	return createSummarizerCore(container, loader, summaryVersion);
+}
+
+describeNoCompat(
+	"Data store realized between startSummary and summarize",
+	(getTestObjectProvider) => {
+		let provider: ITestObjectProvider;
+
+		const createContainer = async (): Promise<IContainer> => {
+			return provider.createContainer(runtimeFactory);
+		};
+
+		async function waitForSummary(summarizer: ISummarizer, refreshLatestAck?: boolean) {
+			const summaryResult = await summarizeNow(summarizer, "test", refreshLatestAck);
+			return summaryResult;
+		}
+
+		beforeEach(async () => {
+			provider = getTestObjectProvider({ syncSummarizer: true });
+		});
+
+		itExpects(
+			"NodeDidNotRunGC error",
+			[
+				{ eventName: "fluid:telemetry:SummarizerNode:NodeDidNotRunGC" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:Summarize_cancel" },
+				{ eventName: "fluid:telemetry:Summarizer:Running:gcUnknownOutboundReferences" },
+			],
+			async () => {
+				const container = await createContainer();
+				await waitForContainerConnection(container);
+				const rootDataObject = await requestFluidObject<RootTestDataObject>(container, "/");
+				const dataObject = await dataStoreFactory1.createInstance(
+					rootDataObject.containerRuntime,
+				);
+				rootDataObject._root.set("store", dataObject.handle);
+				const { container: summarizingContainer, summarizer } = await createSummarizer(
+					provider,
+					container,
+				);
+
+				// This should not fail
+				await assert.rejects(
+					async () => {
+						await provider.ensureSynchronized();
+						await waitForSummary(summarizer);
+					},
+					(error) => {
+						return error.message === "NodeDidNotRunGC";
+					},
+					"expected NodeDidNotRunGC",
+				);
+
+				const deferredAck = new Deferred();
+				summarizingContainer.on("op", (op) => {
+					if (op.type === MessageType.SummaryAck) {
+						deferredAck.resolve(op);
+					}
+				});
+				await provider.ensureSynchronized();
+				await deferredAck.promise;
+				await (summarizer as Summarizer).refreshAndSummarizeLock;
```

Comment:
Why is this needed?

Reply:
Removed, but it was needed for tinylicious tests to complete. The summarizeLock was causing `summarizeOnDemand` to fail